controller/src/beerocks/master/tasks/dynamic_channel_selection_task.cpp

```diff
@@ -58,63 +120,40 @@ void dynamic_channel_selection_task::work()
         LOG(TRACE) << "TRIGGER_SCAN, mac=" << m_radio_mac << ", scan_type is "
                    << ((m_is_single_scan) ? "single-scan" : "continuous-scan");
 
-        // When a scan is requested send the scan parameters Channel pool & Dwell time
-        // Before sending the request set the scan_in_progress flag to true
-        // So another scan would not trigger on the same radio simultaneously
-
+        // Before sending any request, set the scan_in_progress flag to true
+        // So another scan request would not launch on the same radio simultaneously
         database.set_channel_scan_in_progress(m_radio_mac, true, m_is_single_scan);
```

Comment:
Should you first check that this flag is not already `true`?

Reply:
You cannot reach this state if the flag is already `true`